### PR TITLE
Migrate Vendor Classes. #464

### PR DESCRIPTION
**Pull request by jaymckinney-enavate**

### CODE REVIEW

File: Apps/W1/HybridGP/app/src/Migration/Support/HelperFunctions.codeunit.al [hunk 1]
```diff
@@ -1801,7 +1833,10 @@ Codeunit 4037 "Helper Functions"
         if not OpenPurchaseOrdersCreated() then
             CreateOpenPOs();
 
-        if CheckBooksCreated() and OpenPurchaseOrdersCreated() then
+        if not VendorClassesCreated() then
+            CreateVendorClasses();
+
+        if CheckBooksCreated() and OpenPurchaseOrdersCreated() and VendorClassesCreated() then
```
Comment (ToddE123) [inline Apps/W1/HybridGP/app/src/Migration/Support/HelperFunctions.codeunit.al:1839]: These are safe to run premigration and not PostMigration?

File: Apps/W1/HybridGP/app/src/Migration/Vendors/GPVendorMigrator.codeunit.al [hunk 2]
```diff
@@ -8,6 +8,7 @@ codeunit 4022 "GP Vendor Migrator"
         VendorBatchNameTxt: Label 'GPVEND', Locked = true;
         SourceCodeTxt: Label 'GENJNL', Locked = true;
         PostingGroupDescriptionTxt: Label 'Migrated from GP', Locked = true;
+        MigrationDefaultGenProdPostingGroupLbl: Label 'GP', Locked = true;
```
[the remark this answers is not in the document]
Comment (jaymckinney-enavate) [reply Apps/W1/HybridGP/app/src/Migration/Vendors/GPVendorMigrator.codeunit.al:11]: Nope, I didn't see "PostingGroupCodeTxt". Thanks!

File: Apps/W1/HybridGP/app/src/Migration/Vendors/GPVendorMigrator.codeunit.al [hunk 3]
```diff
@@ -356,4 +357,90 @@ codeunit 4022 "GP Vendor Migrator"
...
+        MigrateVendorClasses: Boolean;
+    begin
+        if not GPPM00200.FindSet() then
+            exit;
+
+        MigrateVendorClasses := true;
+        if GPCompanyAdditionalSettings.Get(CompanyName()) then
+            MigrateVendorClasses := GPCompanyAdditionalSettings."Migrate Vendor Classes";
+
+        if MigrateVendorClasses then begin
```
Comment (ToddE123) [inline Apps/W1/HybridGP/app/src/Migration/Vendors/GPVendorMigrator.codeunit.al:379]: why not change to if NOT statement and remove more nesting?

File: Apps/W1/HybridGP/app/src/Migration/Vendors/GPVendorMigrator.codeunit.al [hunk 4]
```diff
@@ -356,4 +356,90 @@ codeunit 4022 "GP Vendor Migrator"
...
+            repeat
```
Comment (ToddE123) [inline Apps/W1/HybridGP/app/src/Migration/Vendors/GPVendorMigrator.codeunit.al:380]: No variables need to be cleared or other for each loop?

**Review by ToddE123 — APPROVED** [review]

**Pull request closed without merging**